crates/wash-lib/src/build.rs

```diff
@@ -96,19 +98,57 @@ pub fn build_actor(
     common_config: &CommonConfig,
     signing_config: Option<SignConfig>,
 ) -> Result<PathBuf> {
-    // Build actor based on language toolchain
-    let actor_wasm_path = match language_config {
-        LanguageConfig::Rust(rust_config) => {
-            build_rust_actor(common_config, rust_config, actor_config)
-        }
-        LanguageConfig::TinyGo(tinygo_config) => {
-            build_tinygo_actor(common_config, tinygo_config, actor_config)
+    let actor_wasm_path = if let Some(raw_command) = actor_config.build_command.as_ref() {
+        // Change directory into the project directory
+        std::env::set_current_dir(&common_config.path)?;
+        let (command, args) = parse_custom_command(raw_command)?;
+        let mut command = process::Command::new(command);
+        // All remaining elements of the split command are interpreted as arguments
+        command.args(args).status().map_err(|e| {
+            if e.kind() == ErrorKind::NotFound {
+                anyhow!("`{:?}` was not found", command.get_program())
+            } else {
+                anyhow!(format!("failed to run `{:?}`: {e}", command.get_program()))
+            }
+        })?;
+
+        let actor_path = actor_config
+            .build_artifact
+            .clone()
+            .map(|p| {
+                // This outputs the path if it's absolute, or joins it with the project path if it's relative
+                if p.is_absolute() {
+                    p
+                } else {
+                    common_config.path.join(p)
+                }
+            })
+            .unwrap_or_else(|| {
+                common_config
+                    .path
+                    .join(format!("build/{}.wasm", common_config.wasm_bin_name()))
+            });
+        if let Err(_e) = std::fs::metadata(actor_path.as_path()) {
+            warn!(
+                "Actor built with custom command but not found in expected path [{}]",
+                actor_path.display()
+            );
         }
-    }?;
+        actor_path
+    } else {
+        // Build actor based on language toolchain
+        let actor_wasm_path = match language_config {
+            LanguageConfig::Rust(rust_config) => {
+                build_rust_actor(common_config, rust_config, actor_config)
+            }
+            LanguageConfig::TinyGo(tinygo_config) => {
+                build_tinygo_actor(common_config, tinygo_config, actor_config)
+            }
+        }?;
```

Comment:
how do you feel about matching on `raw_comand` *and* the `langauge_config` in a tuple here? And maybe moving the logic in the custom stuff (`(Some(cmd), _) => ...`) to another function? 

We way we could keep this pretty easy-to-understand section here as a `match` statement?

Reply:
Since I wanted to skip the embedding this ended up making it a bit more complicated, but I did refactor a bit into a function instead of keeping this all inline